WikiWatershed/model-my-watershed
# Upgrade to Python 3.8

## Overview

Upgrades the codebase to Python 3.8. We went with 3.8 instead of 3.9 because 3.8 comes with Ubuntu Focal.

Connects #3165

## Testing Instructions

* Would recommend code review to be broken down by commit, since there's a lot of changed lines
* Check out this branch
* Delete your local install 😭 
    ```console
    vagrant destroy -f
    ```
* Bring it back up (this can take a long time)
    ```console
    vagrant up
    ```
* Apply this patch to prepare for data fetching:
```diff
diff --git a/scripts/aws/setupdb.sh b/scripts/aws/setupdb.sh
index 80f92048..637fa3fc 100755
--- a/scripts/aws/setupdb.sh
+++ b/scripts/aws/setupdb.sh
@@ -85,9 +85,10 @@ function download_and_load {
 }
 
 function purge_tile_cache {
-    for path in "${PATHS[@]}"; do
-        aws s3 rm --recursive "s3://tile-cache.${PUBLIC_HOSTED_ZONE_NAME}/${path}/"
-    done
+    # for path in "${PATHS[@]}"; do
+    #     aws s3 rm --recursive "s3://tile-cache.${PUBLIC_HOSTED_ZONE_NAME}/${path}/"
+    # done
+    echo "Skipping"
 }
 
 function create_trgm_indexes {
```
* Populate data (this can take a long time)
    ```console
    vagrant ssh app -c 'cd /vagrant && ./scripts/aws/setupdb.sh -bc'
    vagrant ssh app -c 'cd /vagrant && ./scripts/aws/setupdb.sh -dmpq'
    vagrant ssh app -c 'cd /vagrant && ./scripts/aws/setupdb.sh -sS'
    ```
* Restart your VMs to load the latest services
    ```console
    vagrant reload
    ```
* Go to http://localhost:8000/
* Create a user for yourself (since your old data is gone)
* Test:
  - [x] Analyze
  - [x] Monitor searches
  - [x] Stormwater Runoff Model
  - [x] Watershed Multi-Year Model
